src/main/kotlin/no/skatteetaten/aurora/boober/service/OpenShiftCommandBuilder.kt

```diff
@@ -0,0 +1,112 @@
+package no.skatteetaten.aurora.boober.service
+
+import com.fasterxml.jackson.databind.JsonNode
+import no.skatteetaten.aurora.boober.model.AuroraDeployEnvironment
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.service.openshift.OpenShiftClient
+import no.skatteetaten.aurora.boober.service.openshift.OpenShiftResourceClient
+import no.skatteetaten.aurora.boober.service.openshift.OpenShiftResourceClientConfig.ClientType
+import no.skatteetaten.aurora.boober.service.openshift.OpenShiftResourceClientConfig.TokenSource.API_USER
+import no.skatteetaten.aurora.boober.service.openshift.OpenshiftCommand
+import no.skatteetaten.aurora.boober.service.openshift.OperationType.*
+import no.skatteetaten.aurora.boober.service.resourceprovisioning.ProvisioningResult
+import no.skatteetaten.aurora.boober.utils.openshiftKind
+import no.skatteetaten.aurora.boober.utils.openshiftName
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+
+@Service
+class OpenShiftCommandBuilder(
+        val openShiftClient: OpenShiftClient,
+        val openShiftObjectGenerator: OpenShiftObjectGenerator
+) {
+
+    fun generateProjectRequest(environment: AuroraDeployEnvironment): OpenshiftCommand {
+
+        val projectRequest = openShiftObjectGenerator.generateProjectRequest(environment)
+        return createOpenShiftCommand(environment.namespace, projectRequest, false, false)
+    }
+
+    fun generateNamespace(environment: AuroraDeployEnvironment): OpenshiftCommand {
+        val namespace = openShiftObjectGenerator.generateNamespace(environment)
+        return createOpenShiftCommand(environment.namespace, namespace, true, true)
```

Comment:
Here I would prefer to use namedArguments for the booleans that change default value. Also IMHO it could just be an expression.